scheduler/kubernetes/snaps.md

```diff
@@ -15,109 +15,39 @@ This document will show you how to create snapshots of Portworx volumes and how
-volumesnapshots/mysql-snapshot   2s
+Specify following annotations in the `VolumeSnapshot` spec that you use to create the corresponding snapshot type.
 
-NAME                                                                            AGE
-volumesnapshotdatas/k8s-volume-snapshot-2bc36c2d-227f-11e8-a3d4-5a34ec89e61c    1s
-```
+* __px/pre-snap-command__: STORK will run the command which is given in the value of this annotation before taking the snapshot.
+* __px/post-snap-command__: STORK will run the command which is given in the value of this annotation after taking the snapshot.
 
-The creation of the volumesnapshotdatas object indicates that the snapshot has
-been created. If you describe the volumesnapshotdatas object you can see the
-Portworx Volume Snapshot ID and the PVC for which the snapshot was created.
+The commands will be run in pods using the PVC being snapshotted.
 
-```
-$ kubectl describe volumesnapshotdatas 
-Name:         k8s-volume-snapshot-2bc36c2d-227f-11e8-a3d4-5a34ec89e61c
-Namespace:    
-Labels:       <none>
-Annotations:  <none>
-API Version:  volumesnapshot.external-storage.k8s.io/v1
-Kind:         VolumeSnapshotData
-Metadata:
-  Cluster Name:                   
-  Creation Timestamp:             2018-03-08T03:17:02Z
-  Deletion Grace Period Seconds:  <nil>
-  Deletion Timestamp:             <nil>
-  Resource Version:               29989636
-  Self Link:                      /apis/volumesnapshot.external-storage.k8s.io/v1/k8s-volume-snapshot-2bc36c2d-227f-11e8-a3d4-5a34ec89e61c
-  UID:                            2bc3a203-227f-11e8-98cc-0214683e8447
-Spec:
-  Persistent Volume Ref:
-    Kind:  PersistentVolume
-    Name:  pvc-f782bf5c-20e7-11e8-931d-0214683e8447
-  Portworx Volume:
-    Snapshot Id:  991673881099191762
-  Volume Snapshot Ref:
-    Kind:  VolumeSnapshot
-    Name:  default/mysql-snapshot-2b2150dd-227f-11e8-98cc-0214683e8447
-Status:
-  Conditions:
-    Last Transition Time:  <nil>
-    Message:               
-    Reason:                
-    Status:                
-    Type:                  
-  Creation Timestamp:      <nil>
-Events:                    <none>
-```
-### Creating PVCs from snapshots 
+**Examples**
 
-When you install STORK, it also creates a storage class called stork-snapshot-sc.
-This storage class can be used to create PVCs from snapshots.
+Follow is an example of a cassandra volume snapshot where we run the `nodetool flush` command before triggering the snapshot.
 
-To create a PVC from a snapshot, you would add the
-`snapshot.alpha.kubernetes.io/snapshot` annotation to refer to the snapshot
-name.
-
-For the above snapshot, the spec would like this:
 ```
-apiVersion: v1
-kind: PersistentVolumeClaim
+apiVersion: volumesnapshot.external-storage.k8s.io/v1
+kind: VolumeSnapshot
 metadata:
-  name: mysql-snap-clone
+  name: cassandra-snapshot
   annotations:
-    snapshot.alpha.kubernetes.io/snapshot: mysql-snapshot
+    px/pre-snap-command: "nodetool flush"
```

Comment:
We need more options here regarding whether the command needs to be run on each pod or not.